modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSEntries.tsx

```diff
@@ -794,6 +799,115 @@ const FDSEntries = ({
 		});
 	};
 
+	const RenameFDSEntryModalContent = ({
+		closeModal,
+		itemData,
+		loadData,
+		namespace,
+	}: {
+		closeModal: Function;
+		itemData: FDSEntryType;
+		loadData: Function;
+		namespace: string;
+	}) => {
+		const [fdsEntryNameValue, setFdsEntryNameValue] = useState('');
+
+		function saveFDSEntryRename() {
+			fetch(itemData.actions.update.href, {
+				body: JSON.stringify({
+					externalReferenceCode: itemData.externalReferenceCode,
+					label: fdsEntryNameValue,
+				}),
+				headers: {
+					'Accept': 'application/json',
+					'Content-Type': 'application/json',
+				},
+				method: itemData.actions.update.method,
+			})
+				.then(() => {
+					closeModal();
+
+					openToast({
+						message: Liferay.Language.get(
+							'your-request-completed-successfully'
+						),
+						type: 'success',
+					});
+
+					loadData();
+				})
+				.catch(() =>
+					openToast({
+						message: Liferay.Language.get(
+							'your-request-failed-to-complete'
+						),
+						type: 'danger',
+					})
+				);
+		}
+
+		return (
+			<>
+				<ClayModal.Header>
+					{Liferay.Language.get('rename-dataset')}
+				</ClayModal.Header>
+
+				<ClayModal.Body>
+					<ClayForm.Group>
+						<label htmlFor={`${namespace}fdsRenameInput`}>
+							{Liferay.Language.get('name')}
+						</label>
+
+						<ClayInput
+							id={`${namespace}fdsRenameInput`}
+							onChange={(event) =>
+								setFdsEntryNameValue(event.target.value)
+							}
```

Comment:
(optional) You can use ref here instead of state, for performance, like in [the same element in add modal](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/FDSEntries.tsx#L587-L589).  This is controversial, see https://github.com/liferay-frontend/liferay-portal/pull/3320#discussion_r1190738757